src/Symfony/Component/Console/Input/Input.php

```diff
@@ -171,4 +191,62 @@ public function getStream()
     {
         return $this->stream;
     }
+
+    /**
+     * Returns a stringified representation of the options passed to the command.
+     * The options must NOT be escaped as otherwise passing them to a `Process` would result in them being escaped twice.
+     *
+     * @param string[] $optionNames Names of the options returned. If null, all options are returned. Requested options
+     *                              that either do not exist or were not passed (even if the option has a default value)
+     *                              will not be part of the method output.
+     *
+     * @return list<string>
+     */
+    public function unparse(?array $optionNames = null): array
```

Comment:
This function should be extracted to a dedicated class (SRP) so that its easier to maintain of we need to change it. I don't know for the class name and namespace. The method signature would be:

```php
function format(RawInputInterface $input, ?array $optionNames = null): array { /* ... */ }
```

Note: `webmozarts/console-parallelization` has a class dedicated to serialization, and that's great.

Reply:
I agree on principle, and more generally `Input` and its children are doing too much making them harder to maintain (although thankfully they don't change often, so that alleviates that).

I am not sure it is best to move this now though, because:

- in essence it is the counter-part of `::parse()` which is part of `Input`, hence you would put two very closely related pieces in different places
- `::parse()` is abstract, hence every children implement its own. So in theory you could have different implementations of `::unparse()` to specific to an input implementation.